File: Python/Problems/Problems/problem3379.py
import heapq
import math
import sys

import numpy as np

from utils import print1DArray, print2DArray
from typing import List, Set, Optional
from collections import Counter

from .problem import Problem

class Problem3379(Problem):
    def test(self) -> bool:
        nums = [3, -2, 1, 1]

        expected = [1, 1, 1, 3]

        result = self.constructTransformedArray(nums)

        print1DArray(result)

        return expected == result

    def constructTransformedArray(self, nums: List[int]) -> List[int]:
            return [nums[((i + nums[i]) % len(nums) + len(nums)) % len(nums)] for i in range(len(nums))]

File: C++/Problems/Problems/problem_3379.cpp
#include "problem_3379.h"

bool problem_3379::test() {
    std::vector nums { 3, -2, 1, 1 };

    std::vector expected { 1, 1, 1, 3 };

    auto result = constructTransformedArray(nums);

    print1DVector(result);

    return expected == result;
}

std::vector<int> problem_3379::constructTransformedArray(std::vector<int> &nums) {
    int n = nums.size();
    std::vector<int> result(n);
    for (int i = 0; i < n; ++i)
        result[i] = nums[((i + nums[i]) % n + n) % n];
    return result;
}
